types/animation.d.ts

```diff
@@ -0,0 +1,33 @@
+import { PlayerObject } from './model'
+
+export type Animation = CompositeAnimation | typeof WalkAnimation
```

Comment:
I don't think you should use `WalkAnimation` here.
`WalkAnimation` is a subclass of `Animation`, and `Animation` won't just have `WalkAnimation` as its subclass. In the future, other animations such as  `FlyAnimation` and `SwimAnimation` may also be added.
So I suggest you not use a special case when defining `Animation`. It can be a `CompositeAnimation` or just a function which takes `player : PlayerObject` and `time : number` as it input.